clients/kafka-common/src/main/java/org/eclipse/hono/kafka/client/tracing/KafkaHeaderInjectAdapter.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.kafka.client.tracing;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map.Entry;
+
+import io.opentracing.propagation.TextMap;
+import io.vertx.kafka.client.producer.KafkaHeader;
+
+/**
+ * An adapter for injecting properties as a new {@link KafkaHeader} to a list of Vert.x Kafka producer headers.
+ *
+ */
+public final class KafkaHeaderInjectAdapter implements TextMap {
+
+    private final List<KafkaHeader> headers;
+
+    /**
+     * Creates an adapter for a list of {@link KafkaHeader} objects.
+     *
+     * @param headers The list of {@link KafkaHeader} objects.
```

Comment:
I guess the headers param is mandatory and should therefore be checked for `null`?

Reply:
Right. Good catch, thank you.